kotlin/workflow-rx2-runtime/src/main/java/com/squareup/workflow/rx2/FlatMapWorkflow.kt

```diff
@@ -0,0 +1,83 @@
+import com.squareup.workflow.Workflow
+import com.squareup.workflow.WorkflowHost.Update
+import com.squareup.workflow.runWorkflowTree
+import io.reactivex.Flowable
+import io.reactivex.Observable
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.channels.consume
+import kotlinx.coroutines.reactive.openSubscription
+import kotlinx.coroutines.rx2.rxFlowable
+import kotlin.DeprecationLevel.ERROR
+
+/**
+ * Given a stream of [InputT] values to use as inputs for the top-level [workflow], returns a
+ * [Flowable] that, when subscribed to, will start [workflow] and emit its [Update]s. When the
+ * subscription is disposed, the workflow will be terminated.
+ *
+ * The returned [Flowable] is _not_ multicasted – **every subscription will start a new workflow
+ * session.** It is recommended to use a multicasting operator on the resulting stream, such as
+ * [Flowable.replay], to share the updates from a single workflow session.
+ *
+ * The workflow's logic will run in whatever threading context the source [Flowable] is being
+ * observed on.
+ *
+ * This operator is an alternative to using
+ * [WorkflowHost.Factory][com.squareup.workflow.WorkflowHost.Factory] that is more convenient to
+ * use with a stream of inputs.
+ *
+ * This function operates on [Flowable] instead of [Observable] because the workflow runtime
+ * inherently supports backpressure. [Flowable] supports backpressure, [Observable] does not.
+ * RxJava provides operators to adapt between [Flowable]s and [Observable]s by explicitly specifying
+ * how to use backpressure. By operating on [Flowable], this operator leaves it up to the caller to
+ * specify strategies for handling backpressure, instead of assuming any particular behavior.
+ */
+fun <InputT : Any, OutputT : Any, RenderingT : Any> Flowable<InputT>.flatMapWorkflow(
+  workflow: Workflow<InputT, OutputT, RenderingT>,
+  initialSnapshot: Snapshot? = null
+): Flowable<Update<OutputT, RenderingT>> = GlobalScope.rxFlowable(
+    // We use the Unconfined dispatcher here because we want to execute on whatever Rx scheduler the
+    // source Flowable is being observed on.
+    context = Dispatchers.Unconfined
+) {
+  // Convert the input stream into a channel.
+  openSubscription().consume {
+    runWorkflowTree(
+        workflow = workflow.asStatefulWorkflow(),
+        inputs = this,
+        initialSnapshot = initialSnapshot,
+        onUpdate = ::send
+    )
+  }
+}
+
+/**
+ * Use [Flowable.flatMapWorkflow] instead. This function is just defined to be discoverable when
+ * working with [Observable]s instead of [Flowable]s.
+ *
+ * See [Observable.toFlowable] and [Flowable.toObservable] to convert between stream types.
+ */
+@Suppress("DeprecatedCallableAddReplaceWith", "unused")
+@Deprecated("Use Flowable.flatMapWorkflow instead.", level = ERROR)
+fun Observable<*>.flatMapWorkflow(): Nothing =
+  throw UnsupportedOperationException("Use Flowable.flatMapWorkflow instead.")
```

Comment:
This is dumb, just make it take an additional `BackPressure` argument and do the conversions itself.

Reply:
Actually, I'd just delete this one entirely. No one is going to use this layer except for us. Extra guardrails and extra conveniences here will just encourage people to integrate at the wrong layer, and be that much more for us to maintain.

Reply:
Sure.